src/test/java/io/datanerds/avropatch/serialization/ConcurrencyTest.java

```diff
@@ -0,0 +1,102 @@
+package io.datanerds.avropatch.serialization;
+
+import com.google.common.collect.ImmutableList;
+import io.datanerds.avropatch.Patch;
+import org.hamcrest.CoreMatchers;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+import static io.datanerds.avropatch.operation.matcher.PatchMatcher.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class ConcurrencyTest {
+
+    private static Logger logger = LoggerFactory.getLogger(ConcurrencyTest.class);
+    private static final int NUMBER_OF_THREADS = 200;
+    private static final int NUMBER_OF_PATCHES = 100;
+    private static final int MAX_PATCH_SIZE = 50;
+    private static final List<Patch> PATCHES = Collections.unmodifiableList(generateData());
+
+    private static final CustomTypeSerializer SERIALIZER = new CustomTypeSerializer.Builder()
+            .withArray()
+                .withPrimitives()
+                .withCustomTypes()
+                .nullable()
+                .with(Bimmel.class)
+            .endArray()
+                .withPrimitives()
+                .withCustomTypes()
+                .with(Bimmel.class)
+                .nullable()
+            .build();
+
+    @Test
+    public void customTypeSerializer() throws InterruptedException {
+        CountDownLatch startThreadsLatch = new CountDownLatch(1);
+        CountDownLatch threadsFinishedLatch = new CountDownLatch(NUMBER_OF_THREADS - 1);
+        ExecutorService executorService = Executors.newFixedThreadPool(NUMBER_OF_THREADS);
+
+        List<Future<List<Patch>>> futures = new ArrayList<>();
+        for (int i = 0; i < NUMBER_OF_THREADS; i++) {
+            futures.add(executorService.submit(() -> {
```

Comment:
It might improve readability to give this function a name so we can only do something like `executorService.submit(this::serialize)` or whatever name is appropriate.